### PR TITLE
Move associate_user into OrderContents.

### PR DESCRIPTION
**Pull request by athal7**
* Make OrderContents be the canonical place for associating a user.
* Introduce a deprecation warning for anyone calling associate_user through Order.

### CODE REVIEW

File: frontend/app/controllers/spree/orders_controller.rb [hunk 1]
```diff
@@ -37,10 +37,10 @@ def update
 
     # Shows the current incomplete order from the session
     def edit
-      @order = current_order || Order.new
-      associate_user
-      if stale?(current_order)
-        respond_with(current_order)
+      if @order = current_order
```
Comment (jordan-brough) [inline frontend/app/controllers/spree/orders_controller.rb]: We're not allowing creating a new order on demand anymore it looks like? (no replacement for `|| Order.new`?) Is that OK? (I'm not familiar w/ how this page works)

Comment (athal7) [inline frontend/app/controllers/spree/orders_controller.rb]: it's a bit different in master, but maybe we should still support that here. associate_user now does a 'save' going through contents, but apparently it was returning an unpersisted order previously. we can try to maintain that behavior if we want, though we currently don't use frontend.

Comment (athal7) [inline frontend/app/controllers/spree/orders_controller.rb]: looks like the failing specs rely on the behavior of building an order there. i'll put that behavior back

**Comment by jordan-brough** [comment]
awesome :+1:

**Comment by athal7** [comment]
f8f5c0a

**Pull request closed without merging**